src/js/common/components/DashboardRedirect.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { features } from '../containers/BetaApp';
+
+class DashboardRedirect extends React.Component {
+
+  static defaultProps = {
+    services: []
+  };
+
+  componentWillMount() {
+    // Create a global event listener...
+    document.addEventListener('mousedown', event => {
```

Comment:
Can you help me understand why we need this global listener on all of the links?

If the links to the homepage are normal html and not triggered through react-router (full render), will the logic in `componentWillReceiveProps` suffice?  Are there other edge cases that we need to be aware of?

Reply:
The originally idea was to be able to access the Redux data that we need to determine whether or not to redirect, and do that in the most unobtrusive way. The edge case here is that if they click an index link before the user object is loaded then the global listener won't do anything, so they'll still see the regular homepage.

I just got another idea though, which is to drop this redirect on the index page itself, and cache a "beta dashboard" flag on the session to be able to do quick redirects. I'm gonna go ahead and do the merge here and make those adjustments later to keep things clean.